src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -593,21 +593,67 @@ public static void Fill<T>(ref T refData, nuint numElements, T value)
             if (valueLength == 0)
                 return -1;  // A zero-length set of values is always treated as "not found".
 
-            int index = -1;
-            for (int i = 0; i < valueLength; i++)
+            // For the following paragraph, let:
```

Comment:
(GitHub won't let me comment where I want to...)

I see we special-case byte/char in the IndexOfAny public overload, checking to see if the values span is < a known length, and delegating to helpers just for those lengths as appropriate.  It seems we have such helpers for arbitrary Ts as well, but I don't see us doing such delegation in the non-byte/char case.  Are we and I'm just missing it?  Should we?

Reply:
I saw that as well, but check out the comments in `MemoryExtensions.IndexOfAny<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> values)`. They order the checks specifically to demote cases that could've been handled by the existing n-ary overloads.

Maybe this is a more extreme form of demotion in an effort to push callers toward other overloads? :)

In general I have no qualms with introducing this special-case. It seems straightforward enough. Do you think it's appropriate for this PR? Otherwise I'll file a tracking issue.

Reply:
> Otherwise I'll file a tracking issue.

That's fine, thanks.

> Maybe this is a more extreme form of demotion in an effort to push callers toward other overloads? :)

Yeah, I don't know that we should do it, but it's strange that we're doing it only for some size types.  Maybe it's just disproportionately better for those because of the vectorization employed by the overloads for specific arities?

Reply:
https://github.com/dotnet/runtime/issues/53655 opened.